frames/src/main/java/com/checkout/frames/base/validation/ValidateResult.kt

```diff
@@ -0,0 +1,17 @@
+package com.checkout.frames.base.validation
+
+/**
+ * Class representing a validation result which can be either success or failure.
+ */
+internal sealed class ValidateResult<out S : Any> {
```

Comment:
On the failure, I want input to be updating in the text field, whereas in the headless validation result class its throwing a checkout error. So, required a specific custom class for frames. We will discuss this in the call if it can be improved.